firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexBackfillerDelegate.java

```diff
@@ -0,0 +1,66 @@
+package com.google.firebase.firestore.local;
+
+import androidx.annotation.Nullable;
+import com.google.firebase.firestore.index.IndexEntry;
+
+public class SQLiteIndexBackfillerDelegate implements IndexBackfillerDelegate {
+  private final SQLitePersistence persistence;
+  private final IndexBackfiller indexBackfiller;
+
+  SQLiteIndexBackfillerDelegate(SQLitePersistence persistence) {
+    this.persistence = persistence;
+    this.indexBackfiller = new IndexBackfiller(this);
+  }
+
+  public IndexBackfiller getIndexBackfiller() {
+    return indexBackfiller;
+  }
+
+  @Override
+  public void addIndexEntry() {
+    // TODO: implement adding actual values into table
+    persistence.execute(
+        "INSERT OR IGNORE INTO index_entries ("
+            + "index_id, "
+            + "index_value, "
+            + "uid, "
+            + "document_id, "
+            + "active) VALUES(?, ?, ?, ?)",
+        1,
+        "TEST-BLOB",
+        "sample-uid",
+        "sample-documentId");
+  }
+
+  @Override
+  public void removeIndexEntry() {
+    // TODO: implement removing actual values into table
+    persistence.execute(
+        "INSERT OR IGNORE INTO index_entries ("
+            + "index_id, "
+            + "index_value, "
+            + "uid, "
+            + "document_id, "
+            + "active) VALUES(?, ?, ?, ?)",
+        1,
+        "TEST-BLOB".getBytes(),
+        "sample-uid",
+        "sample-documentId");
+    ;
+  }
+
+  @Override
+  @Nullable
+  public IndexEntry getIndexEntry(int indexId) {
```

Comment:
I don't think this belongs here. You should only be writing. We also need to figure out how to index partial document updates, but even for those I don't think we should have a public API.

Reply:
I take this back if you want to use this for testing - but then this should be marked as such.

Reply:
I'll leave this here for the base prototype, and add more features to it as we build it out.